lib/sync.ts

```diff
@@ -0,0 +1,62 @@
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+
+  SPDX-License-Identifier: EPL-2.0
+
+  Copyright Contributors to the Zowe Project.
+*/
+
+import * as EventEmitter from 'events';
+import { KeyVal } from './clusterManager';
+import { SessionSyncCommand, StorageActionSetAll, StorageSyncCommand, StorageActionSet, StorageActionDeleteAll, StorageActionDelete } from './raft-commands';
+const zluxUtil = require('./util');
+const syncLog = zluxUtil.loggers.utilLogger;
+
+export const syncEventEmitter = new EventEmitter();
+
+export function updateSession(sid: string, session: any): void {
+  const data = { sid, session };
+  const sessionLogEntry: SessionSyncCommand = { type: 'session', payload: data };
+  syncLog.info(`updateSession log entry ${JSON.stringify(sessionLogEntry)}`);
+  syncEventEmitter.emit('session', sessionLogEntry);
+}
+
+export function setAllStorageForPlugin(pluginId: string, dict: KeyVal): void {
+  const action: StorageActionSetAll = { type: 'set-all', data: { pluginId, dict } };
+  const storageLogEntry: StorageSyncCommand = { type: 'storage', payload: action };
+  syncLog.info(`setAllStorageForPlugin log entry ${JSON.stringify(storageLogEntry)}`);
```

Comment:
when this gets merged, this should be debug or not existing, because there might be private information.
Also, I'm not certain but   
`syncLog.debug('setAllStorageForPlugin log entry',storageLogEntry);`
might perform better than
`syncLog.debug(`setAllStorageForPlugin log entry ${JSON.stringify(storageLogEntry)}`);`
If the object is not converted to a string until logging occurs, since you do not want to do processing on a log message that will not be printed.

Reply:
`syncLog.debug('setAllStorageForPlugin log entry',storageLogEntry);` outputs `setAllStorageForPlugin log entry [object Object]`.

I would comment out or hide under `if (this.debug) { ... }`. The messages help only I need a deep debugging.